libbeat/beat/beat.go

```diff
@@ -287,55 +283,43 @@ func (bc *instance) cleanup() error {
 }
 
 // launch manages the lifecycle of the beat and guarantees the order in which
-// the Beater methods are invokes and ensures a a proper exit code is set when
-// an error occurs. The exit flag controls if this method calls os.Exit when
-// it completes.
-func (bc *instance) launch(exit bool) error {
-	var err error
-	if exit {
-		defer func() { exitProcess(err) }()
-	}
+// the Beater methods are invoked. If an error occurs in the lifecycle of the
+// Beat it will be returned.
+func (bc *instance) launch() (err error) {
+	defer func() { err = handleError(err) }()
```

Comment:
Generally it's best to avoid named returns, but there weren't a lot of alternatives because I wanted to handle the error value and print the message after the deferred cleanup function completed.